py-polars/tests/unit/io/test_json.py

```diff
@@ -213,6 +213,12 @@ def test_json_deserialize_9687() -> None:
     assert result.to_dict(False) == {k: [v] for k, v in response.items()}
 
 
+def test_json_infer_schema_length_11148() -> None:
+    response = [{"col1": 1}] * 100 + [{"col1": 1, "col2": 2}] * 50
```

Comment:
Can you make this unittest a bit smaller on memory/compute. We can test do this with 3 elements and assert it fail when we set infer_schema_length=2 and succeeds when it is 3.